### PR TITLE
OM-95960: Round up by incrementing recommended Limit by 1 MB

### PR DESCRIPTION
**Pull request by NiraliKotak**
**Background**
This code change intends to resolve [OM-95960](https://vmturbo.atlassian.net/browse/OM-95960). Kubeturbo recommends memory limit to set if user set limit in the deployment is lower than required based on the calculation 
`(kubeturbo_mem_limit_gb * 0.9 - number_of_pods_in_thousands_in_cluster * 0.24 - 1.4) *  5000`

A recommended limit is calculated and displayed in the kubeturbo logs as a warning to set at least recommended memory limit in the kubeturbo deployment. In the warning message, kubeturbo displayed only 2 decimal digits leading to sometimes setting lower value than required. User might see some error message like below which is confusing
`Cannot calculate items per list API call: the memory limit of kubeturbo 3.00 GB is too low to calculate a reasonable number of items per list API call. Kubeturbo may run into OOM. Consider increasing the memory limit of kubeturbo to at least 3.00 GB.`

**Code change**
This change increments any value by 0.001 Gi (1Mi) and displays the warning with 3 decimal points
Screen Shot 2023-02-06 at 10 46 27 PM

Screen Shot 2023-02-06 at 10 46 13 PM

### CODE REVIEW

File: pkg/discovery/processor/cluster_processor.go [hunk 1]
```diff
@@ -286,10 +288,10 @@ func (p *ClusterProcessor) calculateItemsPerListQuery(podCount int) (int, float6
 		recommendedLimitInGB := (1.0 + DefaultExtraClusterWideUsageInGB +
 			(DefaultCadvisorMetricSizePerPodInGB*DefaultMaxPodsPerNode + DefaultExtraPerNodeUsageInGB) +
 			DefaultGenericMetricSizePerThousandPodsInGB*podsInThousands) / DefaultAutoMemLimitPct
-		return 0, 0, fmt.Errorf("the memory limit of kubeturbo %.2f GB is too low to calculate "+
+		return 0, 0, fmt.Errorf("the memory limit of kubeturbo %.3f Gi is too low to calculate "+
```
[the remark this answers is not in the document]
Comment (NiraliKotak) [reply pkg/discovery/processor/cluster_processor.go:291]: I think all the calculations are based on Gi but the suffix used in the warning messages references to GB(suffix G)

**Review by pankajbatra1 — APPROVED** [review]

File: pkg/discovery/processor/cluster_processor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cmedina003) [reply pkg/discovery/processor/cluster_processor.go:291]: Then lets update the names and comments. `recommendedLimitInGB -> recommendedLimitInGI`

**Review by ading1977 — APPROVED** [review]
Thanks for all the discussions and changes.
Look good to me.

**Review by cmedina003 — APPROVED** [review]
LGTM 👍

**Comment by NiraliKotak** [comment]
Thanks for reviews. I will merge once the updated branch runs the check successfully.

**Pull request merged**